pkg/client/cache/listers.go

```diff
@@ -135,7 +135,9 @@ func (s storeToNodeConditionLister) List() (nodes api.NodeList, err error) {
 		// Get the last condition of the required type
 		for _, cond := range node.Status.Conditions {
 			if cond.Type == s.conditionType {
-				nodeCondition = &cond
+				condCopy := cond
+				nodeCondition = &condCopy
```

Comment:
It definitely looks like the old version was bugged (always pointed to the last loop variable that held the last condition in the array), so this is an improvement  However, the old behavior tried (and failed) to take the last matching condition and this takes the first.  I don't know enough to definitively say which behavior is correct.